src/utils/format_json/open_telemetry.c

```diff
@@ -0,0 +1,278 @@
+ * Copyright (C) 2023       Florian octo Forster
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *   Florian octo Forster <octo at collectd.org>
+ **/
+
+#include "collectd.h"
+
+#include "utils/format_json/format_json.h"
+
+#include <yajl/yajl_gen.h>
+
+#define CHECK(f)                                                               \
+  do {                                                                         \
+    int status = (f);                                                          \
+    if (status != 0) {                                                         \
+      ERROR("format_json: %s failed with status %d", #f, status);              \
+      return status;                                                           \
+    }                                                                          \
+  } while (0)
+
+static int json_add_string(yajl_gen g, char const *str) /* {{{ */
+{
+  if (str == NULL) {
+    CHECK(yajl_gen_null(g));
+    return 0;
+  }
+
+  CHECK(yajl_gen_string(g, (const unsigned char *)str,
+                        (unsigned int)strlen(str)));
+  return 0;
+} /* }}} int json_add_string */
+
+static int key_value(yajl_gen g, label_pair_t label) {
+  CHECK(yajl_gen_map_open(g)); /* BEGIN KeyValue */
+
+  CHECK(json_add_string(g, "key"));
+  CHECK(json_add_string(g, label.name));
+
+  CHECK(json_add_string(g, "value"));
+  CHECK(yajl_gen_map_open(g)); /* BEGIN AnyValue */
+  CHECK(json_add_string(g, "stringValue"));
+  CHECK(json_add_string(g, label.value));
+  CHECK(yajl_gen_map_close(g)); /* END AnyValue */
+
+  CHECK(yajl_gen_map_close(g)); /* END KeyValue */
+  return 0;
+}
+
+static int number_data_point(yajl_gen g, metric_t const *m) {
+  CHECK(yajl_gen_map_open(g)); /* BEGIN NumberDataPoint */
+
+  CHECK(json_add_string(g, "attributes"));
+  for (size_t i = 0; i < m->label.num; i++) {
+    CHECK(key_value(g, m->label.ptr[i]));
```

Comment:
Both of the `number_data_point()` callers do it with` CHECK()`, so same error could be logged up to 4 times.

Reply:
Fixed in 1bd6d8c7220fe9f00e9155f4a8dc4570b200ba08